core/domain/classifier/algorithm_registry.py

```diff
@@ -0,0 +1,104 @@
+# coding: utf-8
+#
+# Copyright 2016 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Registry for classification algorithms/classifiers."""
+
+import itertools
+import feconf
+
+
+class Registry(object):
+    """Registry of all classification algorithms/classifiers."""
+
+    # Dict mapping algorithm ids to instances of the classification algorithms.
+    _classification_algorithms = {}
+
+    @classmethod
+    def custom_import(cls, path):
```

Comment:
Hm, why do we need this? Can't we use the existing pattern in, e.g., core.domain.interaction_registry? I think there's something to be said for maintaining consistency...